cpp/include/cugraph/graph_view.hpp

```diff
@@ -821,6 +839,7 @@ class graph_view_t<vertex_t,
                      std::optional<raft::host_span<vertex_t const>>,
                      std::optional<std::byte> /* dummy */>
     local_sorted_unique_edge_dst_vertex_partition_offsets_{std::nullopt};
+  std::optional<std::vector<graph_mask_view_t<vertex_t, edge_t>*>> mask_{std::nullopt};
```

Comment:
The difference between SG & MG should be encapsulated inside the graph_mask_view_t object. This will end-up replicating vertex mask multiple times if there are multiple edge partitions.